test/unit/uk/gov/hmrc/organisationsmatchingapi/services/CrnMatchingServiceSpec.scala

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package unit.uk.gov.hmrc.organisationsmatchingapi.services
+
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import uk.gov.hmrc.organisationsmatchingapi.models
+import uk.gov.hmrc.organisationsmatchingapi.services.CrnMatchingService
+
+class CrnMatchingServiceSpec extends AnyWordSpec with Matchers {
+
+  trait Fixture {
+    val crnMatchingService = new CrnMatchingService
+  }
+
+  "crnMatchingService" should {
+    "basicMatch" should {
+      "match a basic address" in new Fixture {
+
+        val knownFactsData = models.CrnMatchData("mycrn", "myname", models.Address("foo", "bar", None, None, "code"))
+        val ifData = models.CrnMatchData("mycrn", "myname", models.Address("foo", "bar", None, None, "code"))
+
+        crnMatchingService.basicMatch(knownFactsData, ifData)
+      }
+
+      "fail to match where white space differs" in new Fixture {
+        val knownFactsData = models.CrnMatchData("mycrn ", " myname", models.Address("foo ", "b ar", None, None, "code"))
+        val ifData = models.CrnMatchData("mycrn", "myname", models.Address("foo", "bar", None, None, "code"))
+
+        crnMatchingService.basicMatch(knownFactsData, ifData) shouldBe false
+      }
+
+      "fail to match where punctuation differs" in new Fixture {
+        val knownFactsData = models.CrnMatchData("myc!rn ", " my'name", models.Address("foo .", "b ar", None, None, "code"))
+        val ifData = models.CrnMatchData("mycrn", "myname", models.Address("foo", "bar", None, None, "code"))
+
+        crnMatchingService.basicMatch(knownFactsData, ifData) shouldBe false
+      }
+
+      "fail to match where PO box differs" in new Fixture {
+        val knownFactsData = models.CrnMatchData("myc!rn ", " my'name", models.Address("P.O. Box787", "bar", None, None, "code"))
```

Comment:
remove ! in mycrn to ensure test is only checking PO Box cleanup.

Reply:
@parkeradam - Can do but not really a show stopper as each method builds on the last and the fact it passed means its cleaning up the PO Box correctly.

Reply:
This was the nive to have I mentioned in the genrsal comment, probably shouldve put it in the main comment too.